File: problem/Highest_Altitude/largestAltitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    let alt = 0;
    let best = 0;

    for (let g of gain) {
        alt += g;
        if (alt > best) {
            best = alt;
        }
    }

    return best;
};

File: problem/Highest_Altitude/largestAltitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int alt = 0;
        int best = 0;

        for (int g : gain) {
            alt += g; 
            if (alt > best) {
                best = alt;
            }
        }

        return best;
    }
}